arduino-ide-extension/src/browser/theia/editor/editor-command.ts

```diff
@@ -8,9 +8,9 @@ export class EditorCommandContribution extends TheiaEditorCommandContribution {
     // Workaround for https://github.com/eclipse-theia/theia/issues/8722.
     this.editorPreferences.onPreferenceChanged(
       ({ preferenceName, newValue, oldValue }) => {
-        if (preferenceName === 'editor.autoSave') {
-          const autoSaveWasOnBeforeChange = !oldValue || oldValue === 'on';
-          const autoSaveIsOnAfterChange = !newValue || newValue === 'on';
+        if (preferenceName === 'files.autoSave') {
+          const autoSaveWasOnBeforeChange = !oldValue || oldValue !== 'off';
+          const autoSaveIsOnAfterChange = !newValue || newValue !== 'off';
```

Comment:
The `'on'` used to be a workaround to incorrect defaults in Theia. See [here](https://github.com/eclipse-theia/theia/issues/8722), and please ignore me if it has changed in Theia and I am noise only. Thanks!

Reply:
They aren't really supported, but the preference needs to reflect the new monaco expectations. They all work exactly like the old `'on'` value (saving after a short delay).